pkg/tasks/telemeter.go

```diff
@@ -112,15 +114,47 @@ func (t *TelemeterClientTask) create() error {
 	if err != nil {
 		return errors.Wrap(err, "reconciling Telemeter client Secret failed")
 	}
-
-	dep, err := t.factory.TelemeterClientDeployment()
-	if err != nil {
-		return errors.Wrap(err, "initializing Telemeter client Deployment failed")
-	}
-
-	err = t.client.CreateOrUpdateDeployment(dep)
-	if err != nil {
-		return errors.Wrap(err, "reconciling Telemeter client Deployment failed")
+	{
+		// We want to rollout a new deployment of telemeter whenever the configmap trusted-ca-bundle is updated.
+		// Because we react on all events in the same way, we cannot know when the CA TLS actually
+		// changes, so we do a hash style based rollout, similiar to the prometheus-adapter does.
+		proxyCM, err := t.client.GetConfigmap("openshift-monitoring", "trusted-ca-bundle")
+		if err != nil {
+			// Sometimes the ConfigMap might not be created already, if that is the case we should not
+			// error out.
+			if !apierrors.IsNotFound(err) {
+				return errors.Wrap(err, "failed to get trusted-ca-bundle ConfigMap")
+			}
+		}
+		if proxyCM != nil {
+			proxyCM, err := t.factory.TelemeterConfigmapHash(proxyCM)
+			if err != nil {
+				// TODO: if the error is returned "has no data", we should just continue.
+				return errors.Wrap(err, "failed to initiliaze trusted-ca-bundle-<hash> ConfigMap")
```

Comment:
The `TODO` is correct in my opinion. If the configmap is there, but empty, it can mean two things:
1. Trusted proxy CAs have not been enabled at all cluster-wide.
2. Trusted proxy CAs have been enabled, but didn't propagate yet. In that case, once the configmap is filled up by the cluster networking operator, we will catch that event during the next reconcilian loop (as we watch that configmap) and this will enable mounting it into telemeter.

Reply:
Done by just returning an empty configmap, instead of error. We can log that maybe instead?